### PR TITLE
Bug Fix: Delete feature (Fixes three issues)

### PR DESCRIPTION
**Pull request by CertifiedJoon**
Fixes: #3267 
Fixes: #3306 
Fixes: #3378 

Regarding: #3267 
Some of the z-indices were manged by zIndex parameter, while other components where managed by Portals. Somehow the two conflicted and created a weird mix of zIndices. I have changed popover's portal into simple zindex parameter and parents, sibligns, and cousins accordingly.

Regarding: #3306 
The redirection to deleted chat happened because one of its parent elements were a Link component. Previously, upon pressing delete, stopEvent() was called to temporarily pause redirection, and redirection would resume after the delete process. I have changed the Link component into normal button with onClick redirection, maintaining the stopPropogation behavior.

Regarding: #3378 
The default behavior of AlertDialog would wrap the text. But somehow using portal on it would restrict text-wrap behavior. The text-wrap behavior came back after removal of portal while working on #3267 .

deleting from /chat/{id}
image

deleting from /chat
image

toast on delete
image

### CODE REVIEW

File: website/src/components/Chat/ChatListBase.tsx [hunk 1]
```diff
@@ -79,6 +79,7 @@ export const ChatListBase = memo(function ChatListBase({
         bg: "blackAlpha.400",
       }}
       {...props}
+      zIndex="var(--chakra-zIndices-base)"
```
Comment (notmd) [inline website/src/components/Chat/ChatListBase.tsx:82]: why do we need zIndex on this? Isn't it is the last layer by default?

File: website/src/components/Chat/ChatListItem.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const ChatListItem = ({
     <Button
       // @ts-expect-error error due to dynamically changing as prop
       ref={rootRef}
-      {...(!isEditing ? { as: Link, href: ROUTES.CHAT(chat.id) } : { as: "div" })}
+      // {...(!isEditing ? { as: Link, href: ROUTES.CHAT(chat.id) } : { as: "div" })}
```
Comment (notmd) [inline website/src/components/Chat/ChatListItem.tsx:81]: Please revert this. Using a tag is much more better than JS

**Comment by notmd** [comment]
Thanks, but I believe you are relying on zIndex too much which will be hard to debug in future. I will take a look and get rid of it.

File: website/src/components/Chat/ChatListBase.tsx (hunk 1 above)
Comment (CertifiedJoon) [reply website/src/components/Chat/ChatListBase.tsx]: You are right, I've redacted the said line.

File: website/src/components/Chat/ChatListItem.tsx (hunk 2 above)
Comment (CertifiedJoon) [reply website/src/components/Chat/ChatListItem.tsx]: Done.

**Comment by CertifiedJoon** [comment]
> Thanks, but I believe you are relying on zIndex too much which will be hard to debug in future. I will take a look and get rid of it.

Thank you :)

**Comment by notmd** [comment]
I just push all the fix, please check @CertifiedJoon

**Comment by CertifiedJoon** [comment]
@notmd Thank you for your help! Looks great. I will make sure to use portals more often from now on.

**Pull request merged**